src/domain/entities/BasicDataSet.ts

```diff
@@ -0,0 +1,48 @@
+import { Struct } from "$/domain/entities/generic/Struct";
+import { Id, NamedRef } from "$/domain/entities/Ref";
+
+export interface BasicDataSetAttrs {
+    id: Id;
+    translations: D2Translation[];
+    displayName: string;
+    formType: DataSetFormType;
+    attributeValues: AttributeValue[];
+}
+
+export class BasicDataSet extends Struct<BasicDataSetAttrs>() {
+    constructor(attrs: BasicDataSetAttrs) {
+        super(attrs);
+        validateDataSetIsAllowed(this);
+    }
+}
+
+export function validateDataSetIsAllowed(basicDataSet: BasicDataSet): BasicDataSet {
+    const hasHideAttribute = basicDataSet.attributeValues.some(
+        av => av.attribute.name === "hideInTallySheet" && av.value === "true"
```

Comment:
Can we use the code or the client has only the name defined? In any case, move to d2-metadata.ts

Reply:
Indeed. The client only has name defined